lib: lte_link_control: ncellmeas: support for GCI search types (#9153)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

    * LED 1 (nRF9160 DK)/Purple LED (Thingy:91) is lit for five seconds indicating that a current location has been acquired by using the ``location get`` command.
    * Overlay files for nRF9160 DK with nRF7002 EK to enable Wi-Fi scanning support.
      With this configuration, you can, for example, obtain the current location by using the ``location get`` command.

>     * Support for new GCI (Global Cell Id) search types for ``link ncellmeas`` command, which are supported by the modem firmware versions >= 1.3.4.

Reviewer: ```suggestion
    * Support for new GCI (Global Cell ID) search types for ``link ncellmeas`` command, which are supported by the modem firmware versions >= 1.3.4.
```

  * Added timeout for the entire location request.

* :ref:`lte_lc_readme` library:

>   * Added support for GCI (Global Cell Id) neighbor cell measurement search types which are supported by the modem firmware versions >= 1.3.4.
>     Parameter type in :c:func:`lte_lc_neighbor_cell_measurement` changed to :c:struct:`lte_lc_ncellmeas_params` and is
>     including both search type and GCI count that has impact only with GCI search types.

Reviewer: ```suggestion
  * Added support for GCI (Global Cell ID) neighbor cell measurement search types, which are supported by the modem firmware versions >= 1.3.4.
  
    Parameter type in the :c:func:`lte_lc_neighbor_cell_measurement` function changed to :c:struct:`lte_lc_ncellmeas_params`.
    It includes both the search type and GCI count that have an impact only on GCI search types.
```

Author: Changed
